src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/TempPE/TempPEBuildManager.cs

```diff
@@ -0,0 +1,315 @@
+            var objPath = await property.IntermediatePath.GetValueAsPathAsync(false, false);
+            var outputPath = Path.Combine(basePath, objPath, "TempPE");
+
+            var files = new HashSet<string>(inputs.SharedInputs.Count + 1, StringComparers.Paths);
+            files.AddRange(inputs.SharedInputs.Select(i => Path.Combine(basePath, i)));
+            files.Add(Path.Combine(basePath, fileName));
+
+            var outputFileName = fileName + ".dll";
+
+            // var result = await _compiler.CompileAsync(_languageServiceHost.ActiveProjectContext, Path.Combine(outputPath, outputFileName), files, token);
+            //
+            // if (!result)
+            // {
+            //     return null;
+            // }
+
+            // VSTypeResolutionService is the only consumer, and it only uses the codebase element so just default most of them
+            return $@"<root>
+  <Application private_binpath = ""{outputPath}""/>
+  <Assembly
+    codebase = ""{outputFileName}""
+    name = ""{fileName}""
+    version = ""0.0.0.0""
+    snapshot_id = ""1""
+    replaceable = ""True""
+  />
+</root>";
+        }
+
+        protected override Task ApplyAsync(DesignTimeInputsDelta value)
+        {
+            var applied = AppliedValue.Value;
+            var designTimeInputs = applied.Inputs.ToBuilder();
+            var designTimeSharedInputs = applied.SharedInputs.ToBuilder();
+
+            var addedDesignTimeInputs = new List<string>();
+            var removedDesignTimeInputs = new List<string>();
+
+            foreach (var item in value.AddedItems)
+            {
+                // If a property changes we might be asked to add items that already exist, so we just ignore them as we're happy using our existing CancellationSeries
+                if (!designTimeInputs.ContainsKey(item))
+                {
+                    designTimeInputs.Add(item, new CancellationSeries());
+                    addedDesignTimeInputs.Add(item);
+                }
+            }
+
+            foreach (var item in value.AddedSharedItems)
+            {
+                designTimeSharedInputs.Add(item);
+            }
+
+            foreach (var item in value.RemovedItems)
+            {
+                if (designTimeInputs.TryGetValue(item, out var series))
+                {
+                    series?.Dispose();
+                    designTimeInputs.Remove(item);
+                    removedDesignTimeInputs.Add(item);
+                }
+                designTimeSharedInputs.Remove(item);
+            }
+
+            AppliedValue = new ProjectVersionedValue<DesignTimeInputsItem>(new DesignTimeInputsItem
+            {
+                Inputs = designTimeInputs.ToImmutable(),
+                SharedInputs = designTimeSharedInputs.ToImmutable(),
+            }, value.DataSourceVersions);
+
+            // Fire off the events
+            var buildManager = BuildManager.First().Value as VSBuildManager;
```

Comment:
We have a hard dependency on the BuildManager being; BuildManagerEvents and VSBuildManager. Please just import that.

Reply:
Can you expand on that? Do you mean directly importing VSBuildManager (need to add another Export attribute to it?) or do you mean importing VSProjectEvents and access through BuildManagerEvents so that we're ensuring our dependency is hooked up correctly?